pyembroidery/EmbPattern.py

```diff
@@ -1723,14 +1723,19 @@ def static_write(pattern, filename, settings=None):
         """Writes file, assuming type by extension"""
         extension = EmbPattern.get_extension_by_filename(filename)
         extension = extension.lower()
+        supported_extensions = [file_type["extension"] for file_type in EmbPattern.supported_formats()]
 
-        for file_type in EmbPattern.supported_formats():
-            if file_type["extension"] != extension:
-                continue
-            writer = file_type.get("writer", None)
-            if writer is None:
-                continue
+        if extension not in supported_extensions:
+            raise Exception(f"Conversion to file type '{extension}' is not supported")
```

Comment:
This should probably raise either a custom error or a value error. The filename was right but it's value was not among the supported types. It's a bit hard to catch a bare exception without various tools yelling at you.

Also, weirdly the code is so old that the 3.6 fstrings are not supported. I think the code is compatible even back to like 3.4 or so and may even still by Python 2 compatible even though overt support was dropped.